domain/src/test/java/org/oppia/domain/topic/StoryProgressControllerTest.kt

```diff
@@ -51,102 +67,80 @@ class StoryProgressControllerTest {
 
   @Inject lateinit var profileTestHelper: ProfileTestHelper
 
-  @Before
-  fun setUp() {
-    setUpTestApplicationComponent()
-  }
+  @Mock lateinit var mockRecordProgressObserver: Observer<AsyncResult<Any?>>
+  @Captor lateinit var recordProgressResultCaptor: ArgumentCaptor<AsyncResult<Any?>>
 
-  @Test
-  fun testGetStoryProgress_validStory_isSuccessful() {
-    val storyProgressLiveData = storyProgressController.getStoryProgress(TEST_STORY_ID_0)
+  private lateinit var profileId: ProfileId
 
-    val storyProgressResult = storyProgressLiveData.value
-    assertThat(storyProgressResult).isNotNull()
-    assertThat(storyProgressResult!!.isSuccess()).isTrue()
-  }
-
-  @Test
-  fun testGetStoryProgress_validStory_providesCorrectChapterProgress() {
-    val storyProgressLiveData = storyProgressController.getStoryProgress(TEST_STORY_ID_0)
+  private val timestamp = Date().time
 
-    val storyProgress = storyProgressLiveData.value!!.getOrThrow()
-    assertThat(storyProgress.chapterProgressCount).isEqualTo(1)
-    assertThat(storyProgress.chapterProgressMap[TEST_EXPLORATION_ID_0]).isEqualTo(NOT_STARTED)
+  private val coroutineContext by lazy {
+    EmptyCoroutineContext + testDispatcher
   }
 
-  @Test
-  fun testGetStoryProgress_validSecondStory_providesCorrectChapterProgress() {
-    val storyProgressLiveData = storyProgressController.getStoryProgress(TEST_STORY_ID_1)
-
-    // The third chapter should be missing prerequisites since chapter prior to it has yet to be completed.
-    val storyProgress = storyProgressLiveData.value!!.getOrThrow()
-    assertThat(storyProgress.chapterProgressCount).isEqualTo(3)
-    assertThat(storyProgress.chapterProgressMap[TEST_EXPLORATION_ID_1]).isEqualTo(NOT_STARTED)
-    assertThat(storyProgress.chapterProgressMap[TEST_EXPLORATION_ID_2]).isEqualTo(NOT_PLAYABLE_MISSING_PREREQUISITES)
-    assertThat(storyProgress.chapterProgressMap[TEST_EXPLORATION_ID_3]).isEqualTo(NOT_PLAYABLE_MISSING_PREREQUISITES)
-  }
+  // https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-test/
+  @ObsoleteCoroutinesApi
+  private val testThread = newSingleThreadContext("TestMain")
```

Comment:
change this to @ObsoleteCoroutinesApi private val testThread = newSingleThreadContext("TestMain")

Reply:
Done